idb/postgres/internal/encoding/encode.go

```diff
@@ -18,6 +21,39 @@ func EncodeJSON(obj interface{}) []byte {
 	return buf
 }
 
+// printableUTF8OrEmpty checks to see if the entire string is a UTF8 printable string.
+// If this is the case, the string is returned as is. Otherwise, the empty string is returned.
+func printableUTF8OrEmpty(in string) string {
+	// iterate throughout all the characters in the string to see if they are all printable.
+	// when range iterating on go strings, go decode each element as a utf8 rune.
+	for _, c := range in {
+		// is this a printable character, or invalid rune ?
+		if c == utf8.RuneError || !unicode.IsPrint(c) {
+			return ""
+		}
+	}
+	return in
+}
+
+// ConvertAssetParams sanitizes asset param string fields before encoding to JSON bytes.
+func ConvertAssetParams(ap types.AssetParamsWithExtra) types.AssetParamsWithExtra {
+	ap.AssetName = printableUTF8OrEmpty(ap.AssetName)
+	ap.UnitName = printableUTF8OrEmpty(ap.UnitName)
+	ap.URL = printableUTF8OrEmpty(ap.URL)
+	// If the string is printable, don't store the encoded version.
+	// This is a nice optimization, and required for backwards compatibility.
+	if len(ap.AssetName) > 0 {
```

Comment:
this logic is backwards? If *NameBytes has become empty, then *Name must become empty, but *NameBytes can exist while *Name has been squelched.

Reply:
@brianolson I think that given that it's the "sanitize" path, it should be ok, but we should populate the `ap.AssetNameBytes` before the `ap.AssetName` assignment above.

Reply:
Fixed.